refactor: :memo: explain how releases work in CHANGELOG (#27)
rostools/template-workshop: template/CHANGELOG.md

Since we follow [Conventional
Commits](https://decisions.seedcase-project.org/why-conventional-commits)
when writing commit messages, we're able to automatically create formal
"releases" of the workshop based on the commit messages. Releases in the
context of workshops are simply snapshots in time of the workshop

> content that we publish to Zenodo for easier discovery, archival, and
> citation purposes. We use

Reviewer: ```suggestion
content. The releases are published to Zenodo for easier discovery, archival, and
citation purposes. We use
```

[Commitizen](https://decisions.seedcase-project.org/why-semantic-release-with-commitizen)
to be able to automatically create these releases, which uses
[SemVar](https://semverdoc.org) as the version numbering scheme.

> Because a new release is created based on the commit message, releases

Reviewer: ```suggestion
Because releases are created based on commit messages, we release

```
